common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -470,6 +470,33 @@ case class InvokerResourceMessage(status: String,
    * Serializes message to string. Must be idempotent.
    */
   override def serialize: String = InvokerResourceMessage.serdes.write(this).compactPrint
+
+  def canEqual(a: Any) = a.isInstanceOf[InvokerResourceMessage]
+
+  override def equals(that: Any): Boolean =
```

Comment:
The data management service compares the message to decide whether to put data or not.
https://github.com/apache/openwhisk/blob/master/common/scala/src/main/scala/org/apache/openwhisk/core/service/DataManagementService.scala#L161

So we need to override the `equals` method to make it compare messages with `==`.